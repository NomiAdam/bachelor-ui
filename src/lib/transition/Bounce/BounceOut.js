import styled, { keyframes } from 'styled-components';
import BaseAnimation from '../BaseAnimation';

const bounceOutAnimation = keyframes`
  20% {
   transform: scale3d(.9, .9, .9);
  }
  50%, 55% {
   opacity: 1;
   transform: scale3d(1.1, 1.1, 1.1);
  }
  to {
   opacity: 0;
   transform: scale3d(.3, .3, .3);
  }
`;

const BounceOut = styled(BaseAnimation)`
  animation-name: ${ bounceOutAnimation };
`;

export default BounceOut;
