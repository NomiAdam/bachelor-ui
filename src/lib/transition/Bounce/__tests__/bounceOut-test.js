import React from 'react';
import { shallow } from 'enzyme';
import BounceOut from '../BounceOut';

describe('BounceOut', () => {
    it('should be defined', () => {
        expect(BounceOut).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<BounceOut/>)).toMatchSnapshot();
    });
});
