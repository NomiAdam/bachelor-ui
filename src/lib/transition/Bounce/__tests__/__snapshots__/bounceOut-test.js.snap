// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BounceOut should match the snapshot 1`] = `
<styled.div
  className="sc-bwzfXH gILiLG"
  delay="0s"
  direction="normal"
  display="block"
  duration="1s"
  fillMode="both"
  iterationCount="1"
  playState="running"
  timingFunction="ease"
/>
`;
